src/server/routes/api/document/index.js

```diff
@@ -1286,15 +1298,78 @@ const tryVerify = async doc => {
  */
 http.post('/', function( req, res, next ){
   const provided = _.assign( {}, req.body );
-  const { paperId } = provided;
+  const { paperId, elements, performLayout, submit, groundEls } = provided;
   const id = paperId === DEMO_ID ? DEMO_ID: undefined;
   const secret = paperId === DEMO_ID ? DEMO_SECRET: uuid();
+  const email = _.get( provided, 'email', true );
+  const fromAdmin = _.get( provided, 'fromAdmin', true );
 
   const setStatus = doc => tryPromise( () => doc.initiate() ).then( () => doc );
   const handleDocCreation = async ({ docDb, eleDb }) => {
     if( id === DEMO_ID ) await deleteTableRows( API_KEY, secret );
     return await createDoc({ docDb, eleDb, id, secret, provided });
   };
+  const addEls = doc => tryPromise( () => doc.fromJson( { elements } ) ).then( () => doc );
+  const handleLayout = doc => {
+    if ( performLayout ) {
+      return tryPromise( () => doc.applyLayout() ).then( () => doc );
+    }
+
+    return doc;
+  };
+  const handleSubmission = doc => {
+    if ( submit ) {
+      return tryPromise( () => doc.submit() ).then( () => doc );
+    }
+
+    return doc;
+  };
+  const handleElGroundings = doc => {
+    const perform = () => {
+      let entities = doc.entities();
+      let intns = doc.interactions();
+
+      let entityPromises = entities.map( entity => {
+        let name = entity.name();
+        let opts = {
+          limit: 1,
+          offset: 0,
+          name
+        };
+
+        return searchGrounding( opts ).then( res => {
```

Comment:
The BioPAX source will provide an `bp:EntityReference`  with `bp:UnificationXref` , in the case of PhosphoSitePlues, some `uniprot knowledgebase`  id.

We'll have to map those rather than search.

Reply:
e.g. 
```
wget --quiet \
  --method POST \
  --header 'accept: application/json' \
  --header 'content-type: application/x-www-form-urlencoded' \
  --body-data 'from=ACC&to=P_ENTREZGENEID&query=P14873&format=tab' \
  --output-document \
  - https://www.uniprot.org/uploadlists/
```

Reply:
@metincansiper & @jvwong, can this be marked as resolved?

Reply:
Looks fine to me. We can always pop in the grounding-search `/map` here instead, now or later @metincansiper